scope.go

```diff
@@ -482,6 +493,43 @@ func (s *scope) fullyQualifiedName(name string) string {
 	return fmt.Sprintf("%s%s%s", s.prefix, s.separator, name)
 }
 
+func (s *scope) uniqueID(name string, tags map[string]string) string {
+	name = s.fullyQualifiedName(name)
+	l := len(name)
+	for k, v := range tags {
+		l += len(k) + len(v)
+	}
+
+	// size the buffer to include separator bytes
+	buf := bytes.NewBuffer(make([]byte, 0, l+2*len(tags)+1))
```

Comment:
Alternatively we can just create a buffer at the beginning of the `Snapshot` function, and pass the buffer to `uniqueID` function to reuse it across calls. This way we don't need to lock it and doesn't need to store it inside the scope either (which affects the space usage of a real scope)